scripts/github_meta.R

```diff
@@ -1,36 +1,55 @@
-# install.packages(c("yaml", "jsonlite", "devtools"))
-# devtools::install_github("cscheid/rgithub")
-# set environment variable GITHUB_TOKEN
+# install.packages(c("yaml", "jsonlite", "rlang", "gh"))
+# set environment variable GITHUB_PAT
 
 suppressMessages(library(yaml))
 suppressMessages(library(jsonlite))
-suppressMessages(library(github))
+suppressMessages(library(gh))
+
+`%||%` <- rlang::`%||%`
 
 message("* setting up context")
-ctx <- create.github.context(personal_token = Sys.getenv("GITHUB_TOKEN"))
 
 message("* loading yaml")
 yml <- yaml.load_file("_config.yml")
+widgets <- yml$widgets
+
+# use only packages on GitHub
+idx <- vapply(widgets, function(x) !is.null(x$ghuser), FUN.VALUE = logical(1))
+widgets <- widgets[idx]
+
+available_pkgs <- available.packages()[, "Package"]
 
-meta <- lapply(yml$widgets, function(wdgt) {
+meta <- lapply(widgets, function(wdgt) {
+  # if it's not hosted on GitHub, return 0
+  if (is.null(wdgt$ghuser))
+    return(list(stargazers_count = 0)) 
+  
   message("*** getting meta data for: ", wdgt$ghuser, " ", wdgt$ghrepo)
-  a <- try(get.repository(wdgt$ghuser, wdgt$ghrepo))
-  if(inherits(a, "try-error"))
+  
+  res <- try(gh("GET /repos/:owner/:repo", owner = wdgt$ghuser, repo = wdgt$ghrepo))
+
+  if(inherits(res, "try-error"))
     return(list(stargazers_count = 0))
-  a$content[c("stargazers_count", "open_issues_count", "forks_count", "watchers_count")]
+  
+  Sys.sleep(1)
+  
+  # check if the package is on CRAN
+  res$cran <- wdgt$name %in% available_pkgs
```

Comment:
I added this field in the hope of replacing the `cran` flag hard-coded in YAML, but couldn't find the way. So, this is not used at the moment. Please feel free to remove the field.